vnext/Shared/RuntimeOptions.cpp

```diff
@@ -8,38 +8,104 @@ using std::mutex;
 using std::string;
 
 namespace {
-std::unordered_map<string, int32_t> g_runtimeOptions;
+std::unordered_map<string, int32_t> g_runtimeOptionInts;
+std::unordered_map<string, string> g_runtimeOptionStrings;
 mutex g_runtimeOptionsMutex;
 } // namespace
 
 namespace Microsoft::React {
 
 void __cdecl SetRuntimeOptionBool(string &&name, bool value) noexcept {
   lock_guard<mutex> guard{g_runtimeOptionsMutex};
-  g_runtimeOptions.insert_or_assign(std::move(name), value ? 1 : 0);
+  g_runtimeOptionInts.insert_or_assign(std::move(name), value ? 1 : 0);
 }
 
 void __cdecl SetRuntimeOptionInt(string &&name, int32_t value) noexcept {
   lock_guard<mutex> guard{g_runtimeOptionsMutex};
-  g_runtimeOptions.insert_or_assign(std::move(name), value);
+  g_runtimeOptionInts.insert_or_assign(std::move(name), value);
 }
 
 const bool __cdecl GetRuntimeOptionBool(const string &name) noexcept {
   lock_guard<mutex> guard{g_runtimeOptionsMutex};
-  auto itr = g_runtimeOptions.find(name);
-  if (itr != g_runtimeOptions.end())
+  auto itr = g_runtimeOptionInts.find(name);
+  if (itr != g_runtimeOptionInts.end())
     return itr->second == 1;
 
   return false;
 }
 
 const int32_t __cdecl GetRuntimeOptionInt(const string &name) noexcept {
   lock_guard<mutex> guard{g_runtimeOptionsMutex};
-  auto itr = g_runtimeOptions.find(name);
-  if (itr != g_runtimeOptions.end())
+  auto itr = g_runtimeOptionInts.find(name);
+  if (itr != g_runtimeOptionInts.end())
     return itr->second;
 
   return 0;
 }
 
 } // namespace Microsoft::React
+
+void __cdecl Microsoft_React_SetRuntimeOptionBool(const char *name, bool value) noexcept {
+  if (!name)
+    return;
+
+  lock_guard<mutex> guard{g_runtimeOptionsMutex};
+  if (value)
+    g_runtimeOptionInts.insert_or_assign(name, value ? 1 : 0);
```

Comment:
>value ? 1 : 0 [](http://example.com/codeflow?start=47&length=13)

It should never work - we can never set it to 0 - if value is false, then we erase it below, right?